src/front/glsl/context.rs

```diff
@@ -1366,6 +1366,64 @@ impl Context {
                     value
                 }
             }
+            HirExprKind::Method {
+                expr: object,
+                ref name,
+                ref args,
+            } if ExprPos::Lhs != pos => {
+                let args = args
+                    .iter()
+                    .map(|e| self.lower_expect_inner(stmt, parser, *e, ExprPos::Rhs, body))
+                    .collect::<Result<Vec<_>>>()?;
+                match name.as_ref() {
+                    "length" => {
+                        if !args.is_empty() {
+                            parser.errors.push(Error {
+                                kind: ErrorKind::SemanticError(
+                                    ".length() doesn't take any arguments".into(),
+                                ),
+                                meta,
+                            });
+                        }
+                        let lowered_array =
+                            self.lower_expect_inner(stmt, parser, object, pos, body)?.0;
+                        let array_type = parser.resolve_type(self, lowered_array, meta)?;
+                        match array_type {
+                            &TypeInner::Array {
+                                base: _,
+                                size: crate::ArraySize::Constant(size),
+                                stride: _,
+                            } => {
+                                let mut array_length = self.add_expression(
+                                    Expression::Constant(size),
+                                    stmt.hir_exprs[expr].meta,
+                                    body,
+                                );
+                                self.conversion(&mut array_length, meta, ScalarKind::Sint, 4)?;
```

Comment:
Use `forced_conversion` in this case because constant array sizes might already be `Sint`
```suggestion
                                self.forced_conversion(&mut array_length, meta, ScalarKind::Sint, 4)?;
```